src/libstd/sys/unix/fs.rs

```diff
@@ -132,6 +139,32 @@ impl FromInner<raw::mode_t> for FilePermissions {
 impl Iterator for ReadDir {
     type Item = io::Result<DirEntry>;
 
+    #[cfg(target_os = "sunos")]
```

Comment:
Er hm, actually on second though it may be best to call `readdir_r` anyway because we *already* have to store the path locally. I guess whatever's easiest really